shopfloor_reception/tests/test_set_quantity.py

```diff
@@ -381,3 +381,137 @@ def test_scan_new_package(self):
+        # And also, the remaining qty is 4.0, then for each move line,
+        # product_uom_qty = line.qty_done + move's remaining_qty
+        for line in move_lines:
+            self.assertEqual(line.product_uom_qty, line.qty_done + 4.0)
+
+        # Now, let the new user finish its work
+        for __ in range(4):
+            new_service.dispatch(
+                "set_quantity",
+                params={
+                    "picking_id": picking.id,
+                    "selected_line_id": new_line.id,
+                    "quantity": new_line.qty_done,
+                    "barcode": new_line.product_id.barcode,
+                },
+            )
+
+        # We should have a qty_done == 10.0 on both moves and move lines
+        lines_qty_done = sum(move_lines.mapped("qty_done"))
+        self.assertEqual(lines_qty_done, 10.0)
+        self.assertEqual(lines_qty_done, move_lines.move_id.quantity_done)
+
+        # And also, the product_uom_qty should be == qty_done == 5.0
+        for line in move_lines:
+            self.assertEqual(line.product_uom_qty, 5.0)
+            self.assertEqual(line.product_uom_qty, line.qty_done)
+
+        # However, if we pick more than move's product_uom_qty, then lines
+        # product_uom_qty isn't updated, in order to be able to display an error
+        # in the frontend
+
+        for __ in range(2):
+            new_service.dispatch(
+                "set_quantity",
+                params={
+                    "picking_id": picking.id,
+                    "selected_line_id": new_line.id,
+                    "quantity": new_line.qty_done,
+                    "barcode": new_line.product_id.barcode,
+                },
+            )
+
+        # We're 2 over move's product_uom_qty
+        lines_qty_done = sum(move_lines.mapped("qty_done"))
+        self.assertEqual(lines_qty_done, 12.0)
+        self.assertEqual(lines_qty_done, move_lines.move_id.quantity_done)
+
+        # But line's product_uom_qty hasn't changed and is still 10.0
+        self.assertEqual(sum(move_lines.mapped("product_uom_qty")), 10.0)
+
+        # If we lower by 2 the first move qty done, qty_todo will be updated correctly
+        self.service.dispatch(
+            "set_quantity",
+            params={
+                "picking_id": picking.id,
+                "selected_line_id": selected_move_line.id,
+                "quantity": 2.0,
+                "barcode": selected_move_line.product_id.barcode,
+            },
+        )
+
+        self.assertEqual(selected_move_line.qty_done, 3.0)
+        self.assertEqual(selected_move_line.product_uom_qty, 3.0)
+
+        self.assertEqual(new_line.qty_done, 7.0)
+        self.assertEqual(new_line.product_uom_qty, 7.0)
+
+        # And everything's fine on the move
+        move = move_lines.move_id
+        self.assertEqual(move.product_uom_qty, move.quantity_done)
+        self.assertEqual(move.product_uom_qty, 10.0)
```

Comment:
need one more test:
What happens if move quantity_done == move.product_uom_qty, and some other user starts working on this?

Reply:
A new move line with product_uom_qty=0 will be created here https://github.com/OCA/wms/blob/380408f11c56caded17e753306cbc1eea303eb61/shopfloor_reception/services/reception.py#L264
Seems already fine as user will see in red that the quantity_done > 0